### PR TITLE
Added typedefs

### PR DESCRIPTION
**Pull request by danhab99**
Added typescript compatability in `./index.d.ts`

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+declare module 'robots-parser';
+
+interface Robot {
+  isAllowed(url: string, ua?: [string]): boolean | undefined;
+  isDisallowed(url: string, ua?: [string]): boolean | undefined;
+  getMatchingLineNumber(url: string, ua?: [string]): number | undefined;
+  getCrawlDelay(ua?: [string]): number | undefined;
+  getSitemaps(): [string];
```
Comment (AnandChowdhary) [inline index.d.ts:8]: Change `[string]` to `string[]`? 😄

**Review by AnandChowdhary — CHANGES_REQUESTED** [review]

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+declare module 'robots-parser';
+
+interface Robot {
+  isAllowed(url: string, ua?: [string]): boolean | undefined;
```
Comment (samclarke) [inline index.d.ts:4]: Thanks for the PR! Should `[string]` not be `string`?

The UA parameter is optional but should be a string if specified.

**Comment by danhab99** [comment]
I think github is bugging out. @samclarke are you seeing [this commit](https://github.com/danhab99/robots-parser/commit/337fd97f3dc5063ae69f181a6aa6049dfd2f841b) in the pull request?

File: index.d.ts (hunk 2 above)
Comment (samclarke) [reply index.d.ts:4]: Oh the `[ua]` in the docs is meant to mean optional, it's based on [JSDoc optional syntax](https://jsdoc.app/tags-param.html#optional-parameters-and-default-values). I might have to see if there's a clearer naming convention for that as it is a bit confusing.

Yeah, the UA should parameter should be a string. Thanks!

**Comment by samclarke** [comment]
That's weird, it's still showing the previous commit for me. I think Github was having some issues yesterday so it might be related to that?

File: index.d.ts (hunk 2 above)
Comment (danhab99) [reply index.d.ts]: it appears github updated the pull request

**Review by samclarke — APPROVED** [review]
Looks good, thanks for the PR!

I'm going to do a bit of testing this weekend just to double-check it all works and then I'll merge it.

**Comment by samclarke** [comment]
Merged, thanks!

**Comment by danhab99** [comment]
@samclarke would you be willing to create a new patch version so that the NPM listing shows that typedefs are included?

**Pull request merged**